controller_manager:
  ros__parameters:
    update_rate: 1000

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

joint_trajectory_controller:
  ros__parameters:
    joints:
      - sp_j0
      - fl_j0
      - fr_j0
      - fl_j1
      - fl_j2
      - fr_j1
      - rl_j0
      - rl_j1
      - rl_j2
      - fr_j2
      - rr_j0
      - rr_j1
      - rr_j2

    command_interfaces:
      - effort

    state_interfaces:
      - position
      - velocity

    gains:
      sp_j0: {p: 30.0, d: 1.0, i: 0.01}
      fl_j0: {p: 30.0, d: 1.0, i: 0.01}
      fr_j0: {p: 30.0, d: 1.0, i: 0.01}
      fl_j1: {p: 30.0, d: 1.0, i: 0.01}
      fl_j2: {p: 30.0, d: 1.0, i: 0.01}
      fr_j1: {p: 30.0, d: 1.0, i: 0.01}
      rl_j0: {p: 30.0, d: 1.0, i: 0.01}
      rl_j1: {p: 30.0, d: 1.0, i: 0.01}
      rl_j2: {p: 30.0, d: 1.0, i: 0.01}
      fr_j2: {p: 30.0, d: 1.0, i: 0.01}
      rr_j0: {p: 30.0, d: 1.0, i: 0.01}
      rr_j1: {p: 30.0, d: 1.0, i: 0.01}
      rr_j2: {p: 30.0, d: 1.0, i: 0.01}

    allow_partial_joints_goal: true

    state_publish_rate: 1000.0